### ISSUE TITLE
Support for cross-title save loading

### ISSUE THREAD

**Issue by emoose** [emoose/xenia#31]
I'd like to look into cross-title saves, the way it's handled in Banjo-Kazooie/Tooie seems pretty straightforward, using XamContentAggregate* functions, and some XEX headers that specify the title IDs it can search for.

I've got some code ready that might let it work, but got no idea how to actually test this out. Does anyone know where I can find Banjo-Kazooie & Tooie saves that can be stop-n-swopped? Tried some saves I found online but couldn't find anything in-game (I don't really know any of the levels or anything though, so maybe I just didn't look in the right place).

If someone could provide saves & a walk-thru to check if stop-n-swop is working I'd appreciate it.

(also, does anyone know any other games besides Banjo/Mass Effect which have cross-title save support?)

**Comment by Etokapa** [comment]
BlazBlue: Continuum Shift Extend is another game that has cross-title save support (with BlazBlue: Continuum Shift).
![BBCSE](https://user-images.githubusercontent.com/31154606/121946243-57c05b80-cd1a-11eb-90d5-f2b9dca5bde7.png)

**Comment by emoose** [comment]
@Etokapa oh cool thanks for the find, looks like it's similar to Banjo, using the XEX_HEADER_ALTERNATE_TITLE_IDS header in the XEX to store the 415707DA title-ID, which I guess is for BlazBlue: Continuum Shift.

Do you happen to have a save for Continuum Shift I could test with by chance?

**Comment by Etokapa** [comment]
Yup! The Gallery showing unlocked CGs and the Story Mode showing different character routes unlocked means the BBCS save data loaded. I can confirm it works on my end too. Thanks for adding support for it so fast!

**Comment by Etokapa** [comment]
Maybe BL2 could support this as well. I'm unable to check on v1.12 because it always freezes for me when checking for DLC (none is present).
https://www.eurogamer.net/articles/2012-07-07-borderlands-2-rewards-for-borderlands-1-players
From master:
![image](https://user-images.githubusercontent.com/31154606/121972588-0e373700-cd41-11eb-9291-708c96689129.png)

**Comment by emoose** [comment]
Ah think I found the issue with Borderlands 2 - looks it's the problem solved by @Gliniak's https://github.com/xenia-project/xenia/commit/a69a0589912cd979ff81549782e0d51a84247165 commit, basically game would call an API until it returned an error, but Xenia's code for the API doesn't actually let it detect any error atm, so it never signals one, making game just run the API over and over.

(I think a proper fix for that would be changing the `XSUCCEEDED(result)` call to `result == X_ERROR_SUCCESS` tho, since `XSUCCEEDED` is for NTSTATUS results, but `result` is a win32 error code instead, so XSUCCEEDED can't actually detect any error with it... I'll change it to that in stfs-writer in a sec)

**Comment by Etokapa** [comment]
Thanks for the update! I can confirm that LP2 is now going ingame _and_ detects save data from other games. It turns out it can see a lot more games than I remember!

```
* Legendary Warrior - Have a Lost Planet save
* Decorated Soldier - Have a Lost Planet Colonies save
* I LOVE LP - Have a save from both Lost Planet and Lost Planet Colonies
* Played the Demo - Have a save from Lost Planet 2 Online Battle demo
* Street Fighter - Have a Street Fighter save
* BSAA - Have a Resident Evil 5 save
* Devil May Cry - Have a Devil May Cry 4 save
* I've Covered Wars - Have a Dead Rising save
* Capcomaniac - Given if 3 types of save-data related Noms de Guerre have been awarded
```

Here's LP2 awarding the player special name tags for detecting my RE5 and LPC saves.

![LP2](https://user-images.githubusercontent.com/31154606/121984433-8fe68f00-cd58-11eb-8465-f95f52441b99.png)

**Comment by emoose** [comment]
Been trying to look into Mass Effect again, but couldn't find any completed saves online... found some that were pretty much at the end of the game, but Xenia has issues rendering ME atm so I couldn't manage to complete it.

If anyone is able to complete ME on x360 (or if you're good enough, on half-rendered Xenia) and upload the finished save somewhere I'd appreciate it.

**Comment by corywormhole** [comment]
> If someone could provide saves & a walk-thru (what to do after loading the save to see if stop-n-swop is working) I'd appreciate it.

Just tested this, you have to create a Banjo-Tooie save file, then launch Banjo-Kazooie and all of the areas containing S&S items will be available. After collecting an egg or a key, exit and launch Tooie, then go to Heggy's Egg Shed near Wooded Hallow and unlockables from the first game can be activated there.

**Comment by corywormhole** [comment]
https://drive.google.com/file/d/1Z3Rf8WKGF5zBiTsqjmTL_DtDw9BJgPX5/view?usp=sharing
https://drive.google.com/file/d/1C0TOdUT7b1VdSBAPza8yoQA9_QxQAxVG/view?usp=sharing

Here are my saves, they both have enough progress to test for yourself if you wish.
